tests/Integration/Behaviour/Features/Context/Domain/OrderReturnFeatureContext.php

```diff
@@ -0,0 +1,128 @@
+     *
+     * @param string $orderReturnStateReference
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function createOrderReturnState(string $orderReturnStateReference): void
+    {
+        $orderReturnState = new OrderReturnState(null, Configuration::get('PS_LANG_DEFAULT'));
+        $orderReturnState->name = 'New order return state';
+        $orderReturnState->add();
+        SharedStorage::getStorage()->set($orderReturnStateReference, $orderReturnState->id);
+    }
+
+    /**
+     * @When I add order return :orderReturnReference from order :orderReference
+     *
+     * @param string $orderReturnReference
+     * @param string $orderReference
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function createOrderReturn(string $orderReturnReference, string $orderReference): void
+    {
+        $orderId = SharedStorage::getStorage()->get($orderReference);
+        $order = new Order($orderId);
+        $orderReturn = new OrderReturn();
+        $orderReturn->id_customer = (int) $order->id_customer;
+        $orderReturn->id_order = $order->id;
+        $orderReturn->question = 'Why?';
+        $orderReturn->state = 1;
+        $orderReturn->add();
+        $orderDetailIds = $quantities = $customizationIds = $customizationQuantityIds = [];
+        foreach ($order->getProducts() as $product) {
+            $orderDetailIds[] = $product['id_order_detail'];
+            $quantities[] = $product['product_quantity'];
+        }
+        $orderReturn->addReturnDetail($orderDetailIds, $quantities, $customizationIds, $customizationQuantityIds);
+        SharedStorage::getStorage()->set($orderReturnReference, $orderReturn->id);
+    }
+
+    /**
+     * @When I change order return :orderReturnReference state to :orderReturnStateReference
+     *
+     * @param string $orderReturnReference
+     * @param string $orderRetunStateReference
+     *
+     * @throws OrderReturnConstraintException
+     */
+    public function updateOrderReturnState(string $orderReturnReference, string $orderRetunStateReference): void
+    {
+        $orderReturnId = SharedStorage::getStorage()->get($orderReturnReference);
+        $orderReturnStateId = SharedStorage::getStorage()->get($orderRetunStateReference);
+
+        $this->getCommandBus()->handle(
+            new UpdateOrderReturnStateCommand(
+                (int) $orderReturnId,
+                (int) $orderReturnStateId
+            )
+        );
+    }
+
+    /**
+     * @Given :orderReturnReference has state :orderReturnStateReference
+     *
+     * @param string $orderReturnReference
+     * @param string $orderReturnStateReference
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function checkOrderReturnState(string $orderReturnReference, string $orderReturnStateReference): void
```

Comment:
Either you should move this to the "legacy" behat context Or you can keep it here but then you should rely on the `GetOrderReturnForEditing` which would be a good thing because it would allow you to also test the `GetOrderReturnForEditingHandler` this way

Reply:
Good point. Will refactor to use order return for editing.